r/R/dataset.R

```diff
@@ -449,6 +466,10 @@ ScannerBuilder <- R6Class("ScannerBuilder", inherit = Object,
       dataset___ScannerBuilder__UseThreads(self, threads)
       self
     },
+    BatchSize = function(batch_size) {
```

Comment:
Why exactly is this method added in this PR? Doesn't seem related to ParquetFileFormat options. Also, it's not tested.

Reply:
This is a property on the parquet reader so initially I had in alongside the other options in this PR. However it's not a parquet specific concern and it's one that might need tweaking on a scan-by-scan basis so I extracted it to ScanOptions.

[It's tested in C++](https://github.com/apache/arrow/pull/6235/files#diff-1d1911b9db0a34349b3a00cfe24f3ddaR69), but I can certainly add an equivalent one in R